pages/home.js

```diff
@@ -51,64 +51,69 @@ class Home extends React.Component {
         }
       ]
     };
-    this.generateStars=this.generateStars.bind(this)
+    this.goToCoffeeShop = this.goToCoffeeShop.bind(this)
   }
+  async componentDidMount() {
+		this._isMounted = true;
+		try{ 
+      await this.props.getCoffeeShop();
+			console.log(this.props.coffeeShop)
+			if(this._isMounted){
+				this.setState({
+					coffeeShop: this.props.coffeeShop
+				})
+			}
+		}
+		catch(err){
+			console.log(err)
+    }
+  }
+  componentWillUnmount() {
+		this._isMounted = false;
+	}
   goToProfile() {
     Actions.profile();
   }
   goToCoffeeMap() {
     Actions.coffeeMap();
   }
-  goToCoffeeShop() {
-    Actions.coffeeShop();
-  }
-  generateBean(beans){
-    let beanIcons=[]
-    for(let i=0; i<coffeeBeans; i++){
-      beanIcons.push((<Image source={require("./../images/CoffeeBean.png")} style={{ height: 30, width: 30, flexDirection: 'row'}}/>))
+  async goToCoffeeShop(id) {
+    try{ 
+      let select= this.state.cards.filter(shop=>shop.id==id)
+      await this.props.storeCoffeeShop(select[0]);
+      console.log(this.props.coffeeShop);
+      Actions.coffeeShop();
```

Comment:
when you are mapping the thumbnails, you can map the onPress to pass the shop in instead of the id but this will work